examples/topology_optimization/ex_thermo_mech.py

```diff
@@ -75,17 +58,25 @@ def _sensitivity(self, dfdv):
     # Filtering
     s_filtered_variables = fn.append(pym.DensityFilter(s_variables, domain=domain, radius=filter_radius))
 
-    # RAMP with q = 1
+    # RAMP with q = 2
     s_penalized_variables = fn.append(
-        pym.MathGeneral(s_filtered_variables, expression=f"{xmin} + {1 - xmin}*(inp0 / (2 - inp0))"))
+        pym.MathGeneral(s_filtered_variables, expression=f"{xmin} + {1 - xmin}*(inp0 / (3 - 2*inp0))"))
 
-    # Assemble stiffness matrix
-    s_K = fn.append(pym.AssembleStiffness(s_penalized_variables, domain=domain, bc=None))
+    # Assemble stiffness and conductivity matrix
+    s_K = fn.append(pym.AssembleStiffness(s_penalized_variables, domain=domain, bc=prescribed_dofs))
+    s_KT = fn.append(pym.AssemblePoisson(s_penalized_variables, domain=domain, bc=nodes_right))
 
-    # Solve
-    s_force = pym.Signal('f', state=f)
-    s_gravity = fn.append(ThermalExpansionLoading([s_filtered_variables], domain=domain))
+    # Solve for temperature
+    s_heat = pym.Signal('q', state=q)
+    s_temperature = fn.append(pym.LinSolve([s_KT, s_heat]))
 
+    # Determine thermo-mechanical load
+    s_TE = fn.append(pym.ElementAverage(s_temperature, domain=domain))
+    s_xT = fn.append(pym.MathGeneral([s_TE, s_filtered_variables], expression="inp0 * inp1"))
+    s_gravity = fn.append(pym.ThermoMechanical(s_xT, domain=domain, alpha=1.0))
```

Comment:
Why is this variable called gravity?

Reply:
Holdover from @artofscience 's implementation, because both are essentially self load. I'll change the name to something that makes more sense.